code/modules/antagonists/eldritch_cult/eldritch_antag.dm

```diff
@@ -0,0 +1,270 @@
+	to_chat(owner, "Rust - decay and destruction, passive destructive abilities.")
+	to_chat(owner, "Flesh - life and necrosis, allows you to resurrect the dead and summon deadly beasts.")
+	to_chat(owner, "Ash - shadows and secrets, stealth and movement based abilties.")
+	to_chat(owner,"<B>Remember that your power comes from souls!</B>")
+
+/datum/antagonist/ecult/on_gain()
+	var/mob/living/current = owner.current
+	if(ishuman(current))
+		forge_primary_objectives()
+		gain_knowledge(/datum/eldritch_knowledge/spell/basic)
+		gain_knowledge(/datum/eldritch_knowledge/living_heart)
+	current.log_message("has been converted to the cult of the forgotten ones!", LOG_ATTACK, color="#960000")
+	if(!GLOB.reality_smash_track)
+		new /datum/reality_smash_tracker()
+		GLOB.reality_smash_track.Generate(1)
+	GLOB.reality_smash_track.AddMind(owner)
+	START_PROCESSING(SSprocessing,src)
+	if(give_equipment)
+		equip_cultist()
+	. = ..()
+
+/datum/antagonist/ecult/on_removal()
+
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/EK = X
+		EK.on_lose(owner.current)
+
+	if(!silent)
+		owner.current.visible_message("<span class='deconversion_message'>[owner.current] looks like [owner.current.p_theyve()] just reverted to [owner.current.p_their()] old faith!</span>", null, null, null, owner.current)
+		to_chat(owner.current, "<span class='userdanger'>Your mind begins to flare as the otherwordly knowledge escapes your grasp!</span>")
+		owner.current.log_message("has renounced the cult of the old ones!", LOG_ATTACK, color="#960000")
+	GLOB.reality_smash_track.RemoveMind(owner)
+	STOP_PROCESSING(SSprocessing,src)
+
+	. = ..()
+
+/datum/antagonist/ecult/proc/equip_cultist()
+	var/mob/living/carbon/H = owner.current
+	if(!istype(H))
+		return
+	. += ecult_give_item(/obj/item/forbidden_book, H)
+	. += ecult_give_item(/obj/item/living_heart, H)
+
+/datum/antagonist/ecult/proc/ecult_give_item(obj/item/item_path, mob/living/carbon/human/H)
+	var/list/slots = list(
+		"backpack" = ITEM_SLOT_BACKPACK,
+		"left pocket" = ITEM_SLOT_LPOCKET,
+		"right pocket" = ITEM_SLOT_RPOCKET
+	)
+
+	var/T = new item_path(H)
+	var/item_name = initial(item_path.name)
+	var/where = H.equip_in_one_of_slots(T, slots)
+	if(!where)
+		to_chat(H, "<span class='userdanger'>Unfortunately, you weren't able to get a [item_name]. This is very bad and you should adminhelp immediately (press F1).</span>")
+		return FALSE
+	else
+		to_chat(H, "<span class='danger'>You have a [item_name] in your [where].</span>")
+		if(where == "backpack")
+			SEND_SIGNAL(H.back, COMSIG_TRY_STORAGE_SHOW, H)
+		return TRUE
+
+
+/datum/antagonist/ecult/process()
+
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/EK = X
+		EK.on_life(owner.current)
+
+	for(var/X in objectives)
+		if(!istype(X,/datum/objective/stalk))
+			continue
+		var/datum/objective/stalk/S = X
+		if(S.target in view(7,src) && S.target && S.target.current.stat == CONSCIOUS)
+			S.timer -= 1 SECONDS
+	return
```

Comment:
`view` is the expensive proc and should be checked last, if the other conditions are true.

`return` doesn't need to be added at the end of the proc, it can be safely removed.

What is the idea here? Shouldn't the objectives have their own processing proc called?

Reply:
it was easier at the time to just add it onto the antagonist datum, i can probably change that.

Reply:
I will refactor the stalk objective in near future, but for now this works perfectlty fine